### ISSUE TITLE
Pipe would append to a stack trace every time exception is rethrown from Read/FlushAsync

### ISSUE THREAD

**Issue by pakrym** [dotnet/corefx#31388]
Because of incorrect usage of ExceptionDispatchInfo.

https://github.com/dotnet/corefx/issues/31371

**Comment by ahsonkhan** [comment]
@pakrym, can we close this issue based on #31375?

**Comment by pakrym** [comment]
2.1 template

#### Description

Multiple calls to ReadAsync on the faulted pipe would produce exceptions with ever-increasing stacktraces.

#### Customer Impact 
Customers would see exceptions with suspiciously long stacktraces: 

https://github.com/aspnet/KestrelHttpServer/issues/2745
https://github.com/aspnet/SignalR/issues/2672

#### Regression?
No

#### Risk
Only the way we use ExceptionDispatchInfo inside pipe changes. No other logical changes.

**Comment by joshfree** [comment]
@pakrym can you please add the 2.1 PR to this issue and close it once you’ve merged the PR? Checkin needs to be completed before Tuesday to make 2.1.4

**Comment by pakrym** [comment]
Sorry, though this would go into 2.1.5.

**Comment by pakrym** [comment]
@joshfree can we postpone this until 2.1.5? We think this fix introduced a behaviour change.

**Comment by joshfree** [comment]
Yes absolutely.

**Comment by joshfree** [comment]
@muratg @pakrym please mark this servicing-consider when the fix is ready to bring back to shiproom. Thanks.

**Comment by karelz** [comment]
Fixed in release/2.1 branch in PR #31680. Will be part of 2.1.5 release.
